lib/tap.js

```diff
@@ -1,253 +1,34 @@
 'use strict'
-const Test = require('./test.js')
-const Stdin = require('./stdin.js')
-const Spawn = require('./spawn.js')
-const util = require('util')
-const objToYaml = require('./obj-to-yaml.js')
-const _didPipe = Symbol('_didPipe')
-const _unmagicPipe = Symbol('_unmagicPipe')
-const Domain = require('async-hook-domain')
+const settings = require('../settings.js')
```

Comment:
Loading `../settings.js` before `libtap` is mandatory as it edits `libtap/settings` before the singleton is created by `libtap`.  For a user test which wants to edit settings this would happen as follows:
```js
const settings = require('tap/settings')
// Alter settings object
const t = require('tap')
```

If using `libtap` then the above code would use `libtap/settings` and `libtap`.  The following would be wrong:
```js
const settings = require('libtap/settings')
// Alter settings object
const t = require('tap')
```

In this case `require('tap')` would cause `tap/settings` to load for the first time, performing manipulations after the user changes.  This is an edge case as I think most users would not alter the settings.  Project-level alterations to settings can be done with a `--require` script through `--node-arg`:
```js
const settings = require('tap/settings')
// Alter settings object for all tests
```